### DISCUSSION TITLE
关于计时器的问题

### DISCUSSION THREAD

**Discussion by novada0** [SUSTech-CS305-Fall22/CS305-Project-Skeleton#46] [Q&A]
我们要对发送的包计时，但是整个过程又是单线程，没想到如果不用多线程还能怎么计时。

**Comment by TANSixu** [comment]
Hi，一种可能的设计是在主循环里检查每个连接的超时情况，可以往这个方向思考一下噢～

**Reply by Lynchrocket** [reply]
请问是在主循环每次循环检查一次每个连接的超时情况吗？但是如果在主循环每次循环都检查一次不会检查得太频繁了吗，这样会不会消耗很多资源从而减缓效率呢？

**Reply by TANSixu** [reply]
> 请问是在主循环每次循环检查一次每个连接的超时情况吗？但是如果在主循环每次循环都检查一次不会检查得太频繁了吗，这样会不会消耗很多资源从而减缓效率呢？

Hi,这样是可以的。不会消耗很多，对于每个连接每次只需要检查和当前时间的差值就行啦。
